src/v/reflection/type_traits.h

```diff
@@ -29,14 +30,6 @@
 
 namespace detail {
 
-template<typename T, template<typename...> class C>
-struct is_specialization_of : std::false_type {};
```

Comment:
Just curious why this moved and not the others? And why did it need to move?

Reply:
Same with optional. I guess they're the ones that config uses?

Reply:
is_specialization_of moved to `base` because it's a foundational building block of type trait utilities. std::optional moved because its only dependent on `std::`. the others were left alone because they either didn't need to move (yet) or they aren't foundational (e.g. is_fragmented_vector).